packages/runtime/package.json

```diff
@@ -2,113 +2,91 @@
-      "require": "./dist/api.cjs.js"
+      "require": "./dist/cjs/api/index.js",
+      "import": "./dist/esm/api/index.js",
+      "types": "./dist/types/api/index.d.ts"
     },
     "./next": {
-      "import": "./dist/next.es.js",
-      "require": "./dist/next.cjs.js"
+      "require": "./dist/cjs/next/index.js",
+      "import": "./dist/esm/next/index.js",
+      "types": "./dist/types/next/index.d.ts"
+    },
+    "./next/plugin": {
+      "require": "./node_modules/@makeswift/next-plugin/index.js",
+      "import": "./node_modules/@makeswift/next-plugin/index.js",
+      "types": "./node_modules/@makeswift/next-plugin/index.d.ts"
     },
-    "./next/plugin": "./next/plugin.js",
     "./builder": {
-      "import": "./dist/builder.es.js",
-      "require": "./dist/builder.cjs.js"
+      "require": "./dist/cjs/builder/index.js",
+      "import": "./dist/esm/builder/index.js",
+      "types": "./dist/types/builder/index.d.ts"
     },
     "./controls": {
-      "import": "./dist/controls.es.js",
-      "require": "./dist/controls.cjs.js"
+      "require": "./dist/cjs/controls/index.js",
+      "import": "./dist/esm/controls/index.js",
+      "types": "./dist/types/controls/index.d.ts"
     },
     "./slate": {
-      "import": "./dist/slate.es.js",
-      "require": "./dist/slate.cjs.js"
+      "require": "./dist/cjs/slate/index.js",
+      "import": "./dist/esm/slate/index.js",
+      "types": "./dist/types/slate/index.d.ts"
     },
     "./state/breakpoints": {
-      "import": "./dist/state/breakpoints.es.js",
-      "require": "./dist/state/breakpoints.cjs.js"
-    }
-  },
-  "typesVersions": {
-    "*": {
-      "prop-controllers": [
-        "dist/types/src/prop-controllers/index.d.ts"
-      ],
-      "react": [
-        "dist/types/src/react.d.ts"
-      ],
-      "box-model": [
-        "dist/types/src/box-model.d.ts"
-      ],
-      "components": [
-        "dist/types/src/components/index.d.ts"
-      ],
-      "api": [
-        "dist/types/src/api/index.d.ts"
-      ],
-      "next": [
-        "dist/types/src/next/index.d.ts"
-      ],
-      "builder": [
-        "dist/types/src/builder/index.d.ts"
-      ],
-      "controls": [
-        "dist/types/src/controls/index.d.ts"
-      ],
-      "slate": [
-        "dist/types/src/slate/index.d.ts"
-      ],
-      "state/breakpoints": [
-        "dist/types/src/state/modules/breakpoints.d.ts"
-      ]
+      "require": "./dist/cjs/state/module/breakpoints.js",
+      "import": "./dist/esm/state/module/breakpoints.js",
+      "types": "./dist/types/state/module/breakpoints.d.ts"
     }
   },
+  "types": "./dist/types/index.d.ts",
   "scripts": {
-    "dev": "concurrently -k 'tsc --watch --preserveWatchOutput' 'vite build --watch'",
+    "dev": "concurrently -k 'tsc --watch --preserveWatchOutput' 'tsup --watch'",
     "clean": "rm -rf dist",
-    "build": "pnpm run clean && tsc && vite build",
-    "preview": "vite preview",
+    "build": "pnpm run clean && tsc && tsup",
```

Comment:
What do you think about adding the `clean` option to tsup, omitting `rm -rf dist` and swapping `tsup` and `tsc`? e.g.
```
"build": "tsup && tsc"
```

Reply:
I played around with doing stuff like that as well as driving `tsc` from the `tsup` config or using `dts`. Ended up going with this because it was equivalent to what we had before with Vite but we could also do it that way too. It's worth noting, though, that running `tsup` would blast away `tsc`'s output which might be annoying. I'm not sure how often one would run `tsup` by itself instead of running `build`, though. `tsc` is sometimes slow so perhaps quite a bit? 🤷🏻‍♂️

Reply:
I would assume if you are going to rebuild, you'll most likely need to compile the types too, unless you are only modifying internals and not external facing APIs 🤷‍♂️  We could also speed up `tsc` using `incremental`?